honeybee_grasshopper_energy/src/HB Shadow Calculation.py

```diff
@@ -0,0 +1,108 @@
+                calculation will be used to distribute incoming diffuse solar
+                energy between interior surfaces.
+            4 = "FullInteriorAndExteriorWithReflections" - The simulation will
+                perform the solar calculation in a manner that accounts for light
+                bounces that happen both outside and inside the zones.  This is the
+                most accurate method and is the one assigned by default.  Note that,
+                if you use this method, EnergyPlus will give Severe warnings if
+                your zones have concave geometry (aka. are "L"-shaped).  Such
+                geometries are not supported by this solar distribution calculation
+                and it is recommeded that you either break up your zones in this
+                case or not use this solar distribution method.
+        _calc_method_: Text describing how the solar and shading models are calculated
+            with respect to the time of calculations during the simulation.
+            Default - AverageOverDaysInFrequency. Choose from the following:
+                * AverageOverDaysInFrequency
+                * TimestepFrequency
+        _frequency_: Integer for the number of days in each period in
+            which a unique shadow calculation will be performed. This field is only
+            used if the AverageOverDaysInFrequency method is used in the previous
+            field. Default - 30.
+        _max_figures_: Integer for the number of figures used in shadow overlaps.
+            Default - 15000.
+    
+    Returns:
+        shadow_calc: A ShadowCalculation object that can be connected to the
+            "HB Simulation Parameter" component in order to specify settings
+            for the EnergyPlus Shadow Calculation.
+"""
+
+ghenv.Component.Name = "HB Shadow Calculation"
+ghenv.Component.NickName = 'ShadowCalc'
+ghenv.Component.Message = '0.1.0'
+ghenv.Component.Category = "Energy"
+ghenv.Component.SubCategory = '5 :: Simulate'
+ghenv.Component.AdditionalHelpFromDocStrings = "2"
+
+try:
+    from honeybee_energy.simulation.shadowcalculation import ShadowCalculation
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee:\n\t{}'.format(e))
+
+
+# dictionary to convert to acceptable solar distributions
+SOLAR_DISTRIBUTIONS = {
+    '0': 'MinimalShadowing',
+    '1': 'FullExterior',
+    '2': 'FullInteriorAndExterior',
+    '3': 'FullExteriorWithReflections',
+    '4': 'FullInteriorAndExteriorWithReflections',
+    'MinimalShadowing': 'MinimalShadowing',
+    'FullExterior': 'FullExterior',
+    'FullInteriorAndExterior': 'FullInteriorAndExterior',
+    'FullExteriorWithReflections': 'FullExteriorWithReflections',
+    'FullInteriorAndExteriorWithReflections': 'FullInteriorAndExteriorWithReflections'
+    }
+
+
+# set default values
+_solar_dist_ = SOLAR_DISTRIBUTIONS[_solar_dist_] if _solar_dist_ is not None \
+    else 'FullInteriorAndExteriorWithReflections'
```

Comment:
Ok, I will add a `try / except KeyError`. Using `dict.get` seems dangerous since it can give the user a false sense that they have successfully changed the solar distribution when they haven't.